### ISSUE TITLE
`iroha_wasm_builder` progress-bar

### ISSUE THREAD

**Issue by Arjentix** [hyperledger/iroha#3237]
It will be nice to have a way to get information about current progress of build from `iroha_wasm_builder`.

As an example of current implementation you can run `kagami validator` and will see, that it looks like it's stucked.

I believe, there is a way to provide this info from `iroha_wasm_builder` crate and the display it on user side.

**Comment by Arjentix** [comment]
Blocked by #2152

**Comment by 0x009922** [comment]
I use a spinner in `kagami swarm` during the building of the validator:

- #3475

We can probably reuse it in `kagami genesis` and in `iroha_wasm_builder` itself. If the PR will be merged, ofc.

**Comment by 0x009922** [comment]
This issue is partially addressed by

- #3841 

That PR introduces `iroha_wasm_builder_cli`, which displays spinners while building & optimizing. However, **it is not a progress bar**. I guess _this_ issue is about displaying a precise progression, e.g. how many crates were already checked/built.

**Comment by DCNick3** [comment]
1. exposing cargo's output: this will use the native cargo implementation of the progress bar. no longer capturing stdout, so diagnostics go to the console too
2. parsing cargo's progress bar. possible, but requires special handling for the diagnostics: they have to somehow be separated from the progress bar
3. cargo json output: AFAIU it doesn't provide enough information to build the progress bar (we need the total # of jobs)
4. cargo as a library: doesn't help, cargo exposes basically the same API surface as the console API. will either require parsing or reimplementing large parts of the cargo itself. the progress bar is buried in non-public part of the API: https://docs.rs/cargo/latest/src/cargo/core/compiler/job_queue/mod.rs.html#469

I've decided to go with option 1, with possibly switching to 2 if some limitations are to be found
